frontend/controller/actions/group-kv.js

```diff
@@ -46,21 +60,22 @@ export default (sbp('sbp/selectors/register', {
       throw new Error('Unable to update lastLoggedIn without an active session')
     }
 
+    const now = sbp('chelonia/time') * 1000
     if (throttle) {
       const state = sbp('state/vuex/state')
       const lastLoggedInRawValue: ?string = state.lastLoggedIn?.[contractID]?.[identityContractID]
       if (lastLoggedInRawValue) {
         const lastLoggedIn = new Date(lastLoggedInRawValue).getTime()
 
-        if ((Date.now() - lastLoggedIn) < LAST_LOGGED_IN_THROTTLE_WINDOW) return
+        if ((now - lastLoggedIn) < LAST_LOGGED_IN_THROTTLE_WINDOW) return
       }
     }
 
-    const now = new Date().toISOString()
+    const nowString = new Date(now).toISOString()
```

Comment:
Same change needed for all other KV updates?

Reply:
Are there other KV values that are date-based? I see only notifications, which have:

```
timestamp: data.createdDate ? new Date(data.createdDate).getTime() : Date.now(),
```

Reply:
Using `'chelonia/time'` here, which is mostly called from contracts, might need to be `async`, which would make the entire function async. I'd leave it as is an add a comment.

Reply:
> which would make the entire function async

...why? `'chelonia/time'` is sync... not async

Reply:
I deleted the comment after checking where it's imported. But if `chelonia/time` is called from a contract, or from the browser it could be upgraded to async. However, we might need to upgrade contracts to await on calls to emit notifications, or basically anywhere external (i.e., defined outside the contract) selectors are used.